lib/ansible/modules/windows/win_nssm.ps1

```diff
@@ -19,7 +21,7 @@ $name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $tru
 $state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present","absent","started","stopped","restarted" -resultobj $result
 
 $application = Get-AnsibleParam -obj $params -name "application" -type "str"
-$appParameters = Get-AnsibleParam -obj $params -name "app_parameters" -type "str"
+$appParameters = Get-AnsibleParam -obj $params -name "app_parameters"
```

Comment:
I've never used this module before but after looking at the existing UI I'm quite confused. Honestly I think we should look into simplifying it a bit more. My thoughts are;

* `app_parameters` is designed to take in a dict/hashtable according to the docs and that seems wrong
* `app_parameters_free_form` is meant to be a string of parameters which is more what I expect but the name of the option isn't that great
* Ultimately a dict/hashtable is a poor way of defining some args considering the YAML syntax makes it hard to start keys with `-`

I'll leave it up to you whether you want to address this in this change or in a future PR but I think we should do is;

* deprecate the `app_parameters_free_form` option
* have `app_parameters` accept a string, list, or dict/hashtable but deprecate the dict/hashtable form as it is confusing
* potentially change `app_parameters` to be just `arguments` or `parameters` and have `app_parameters` as an alias for this

Let me know your thoughts, obviously you used or have a need for this module so would know more.

Reply:
It's also exactly what I think, as I wrote in my previous comment more succinctly: 
> => I suggest removing app_parameters_free_form, have app_parameters accept both a simple string or a list, as it's done in other modules like win_package, and abandon the dict support (IMHO not really helpful).

However, for the renaming of `app_parameters`, I'm less convinced because this name match the internal name used by nssm, but this can be discussed later:
![image](https://user-images.githubusercontent.com/3837104/44775025-dd2bee80-ab74-11e8-8bbe-17b4314977c3.png)

The goal of this first PR remains to fix the current codebase in the simplest and fastest way so that it can be merged quickly and maybe backported.

I totally agree that this module would require a lot more changes, improvements, and features addition (including check mode support), but I prefer to address this in futures PRs to avoid mixing bug fixes, feature requests and breaking changes and delay this first PR ;)

Reply:
Cool if the nssm refers to it as AppParameters then keeping the name sounds good, I wasn't aware of this. Perfectly fine for all this to be resolved in future PRs just wanted to bring out my concerns with the current UI.

Reply:
As stated in https://github.com/jborean93, let's add an explicit check and error our if the value is a `Hashtable` and not a string. We should also deprecate this option and remove in 2.12.